unsigned int tomo_db0::IoLoop::epoll_pass<false, false, false>(int)
--- PRE

+++ POST

@@ -2,7 +2,7 @@

 0004 [ 2] push %r15
 0006 [ 2] push %r14
 0008 [ 2] push %r13
-000a [ 3] mov %rdi,%r14
+000a [ 3] mov %rdi,%r15
 000d [ 2] push %r12
 000f [ 1] push %rbp
 0010 [ 1] push %rbx
@@ -13,169 +13,161 @@

 0024 [ 5] mov $0x100,%edx
 0029 [ 3] mov %rbx,%rsi
 002c [ 3] xor %r12d,%r12d
-002f [ 9] mov %fs:0x28,%rax
-0038 [ 5] mov %rax,0x28(%rsp)
-003d [ 2] xor %eax,%eax
-003f [ 5] call <epoll_wait@plt>
-0044 [ 8] movl $0x0,0xc(%rsp)
-004c [ 2] test %eax,%eax
-004e [ 4] cmovns %eax,%r12d
-0052 [ 7] mov %r12d,0xc18(%r14)
-0059 [ 6] jle <+0x11f>
-005f [ 7] mov 0x190(%r14),%rax
-0066 [ 3] movslq %r12d,%rdx
-0069 [ 2] xor %ebp,%ebp
-006b [ 7] lea <rip>(%rip),%r15 # <local-switch-0x110,0x250,0x1d0,0x110,0x110,0x180,0x110,0x110,0x100,0x110,0x110,0x110,0x110,0x110,0x110,0x110,0x110,0x150>
-0072 [ 7] add %rdx,0x520(%rax)
-0079 [ 7] lea 0xac8(%r14),%rax
-0080 [ 5] mov %rax,0x10(%rsp)
-0085 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
-0097 [ 4] shr $0x38,%rax
-009b [ 2] cmp $0x11,%al
-009d [ 2] ja <+0x110>
-009f [ 4] movslq (%r15,%rax,4),%rax
-00a3 [ 3] add %r15,%rax
-00a6 [ 3] notrack jmp *%rax
-00a9 [ 2] mov (%rbx),%eax
-00ab [ 5] test $0x2019,%eax
-00b0 [ 2] je <+0xb6>
-00b2 [ 4] movb $0x0,0x29(%rdx)
-00b6 [ 2] test $0x1c,%al
-00b8 [ 2] je <+0xf2>
-00ba [ 4] cmpb $0x0,0x30(%rdx)
-00be [ 5] mov %rdx,0x20(%rsp)
-00c3 [ 2] jne <+0xf2>
-00c5 [ 7] mov 0x1d8(%r14),%rcx
-00cc [ 7] mov 0x1c8(%r14),%rax
-00d3 [ 4] movb $0x1,0x30(%rdx)
-00d7 [ 4] sub $0x8,%rcx
-00db [ 3] cmp %rcx,%rax
-00de [ 6] je <+0x2ea>
-00e4 [ 3] mov %rdx,(%rax)
-00e7 [ 4] add $0x8,%rax
-00eb [ 7] mov %rax,0x1c8(%r14)
-00f2 [ 3] mov %rdx,%rsi
-00f5 [ 3] mov %r14,%rdi
-00f8 [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
-00fd [ 3] nopl (%rax)
-0100 [ 4] incl 0xc(%rsp)
-0104 [11] data16 cs nopw 0x0(%rax,%rax,1)
-010f [ 1] nop
-0110 [ 2] inc %ebp
-0112 [ 4] add $0xc,%rbx
-0116 [ 3] cmp %r12d,%ebp
-0119 [ 6] jl <+0x90>
-011f [ 5] mov 0x28(%rsp),%rax
-0124 [ 9] sub %fs:0x28,%rax
-012d [ 6] jne <+0x2e5>
-0133 [ 4] mov 0xc(%rsp),%eax
-0137 [ 4] add $0x38,%rsp
-013b [ 1] pop %rbx
-013c [ 1] pop %rbp
-013d [ 2] pop %r12
-013f [ 2] pop %r13
-0141 [ 2] pop %r14
-0143 [ 2] pop %r15
-0145 [ 1] ret
-0146 [10] cs nopw 0x0(%rax,%rax,1)
-0150 [ 7] mov 0x190(%r14),%rax
-0157 [ 5] mov 0x10(%rsp),%rdx
-015c [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
-0163 [ 2] inc %ebp
-0165 [ 4] add $0xc,%rbx
-0169 [ 2] mov (%rax),%esi
-016b [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
-0170 [ 4] incl 0xc(%rsp)
-0174 [ 3] cmp %r12d,%ebp
-0177 [ 6] jl <+0x90>
-017d [ 2] jmp <+0x11f>
-017f [ 1] nop
-0180 [ 5] lea 0x20(%rsp),%r13
-0185 [ 9] movq $0x0,0x20(%rsp)
-018e [ 2] xchg %ax,%ax
-0190 [ 7] mov 0xbc8(%r14),%edi
-0197 [ 5] mov $0x8,%edx
-019c [ 3] mov %r13,%rsi
-019f [ 5] call <read@plt>
-01a4 [ 4] cmp $0x8,%rax
-01a8 [ 2] je <+0x190>
-01aa [ 7] mov 0x190(%r14),%rax
-01b1 [ 2] inc %ebp
-01b3 [ 4] add $0xc,%rbx
-01b7 [ 4] incl 0xc(%rsp)
-01bb [ 7] incq 0x440(%rax)
-01c2 [ 3] cmp %r12d,%ebp
-01c5 [ 6] jl <+0x90>
-01cb [ 5] jmp <+0x11f>
-01d0 [10] movabs $0xffffffffffff,%rax
-01da [ 3] and %rax,%rdx
-01dd [ 6] je <+0x110>
-01e3 [ 6] mov 0x788(%rdx),%eax
-01e9 [ 7] mov 0x190(%r14),%rcx
-01f0 [ 2] cmp %eax,(%rcx)
-01f2 [ 6] jne <+0x110>
-01f8 [ 4] cmpb $0x0,0x33(%rdx)
-01fc [ 6] jne <+0x110>
-0202 [ 6] mov 0x784(%rdx),%eax
-0208 [ 3] cmp $0xfffffffe,%eax
-020b [ 6] je <+0x110>
-0211 [ 7] mov 0x288(%r14),%rax
-0218 [ 7] mov 0x290(%r14),%rcx
-021f [ 3] cmp %rcx,%rax
-0222 [ 2] jne <+0x23d>
-0224 [ 5] jmp <+0xa9>
-0229 [ 7] nopl 0x0(%rax)
-0230 [ 4] add $0x28,%rax
-0234 [ 3] cmp %rax,%rcx
-0237 [ 6] je <+0xa9>
-023d [ 3] cmp (%rax),%rdx
-0240 [ 2] jne <+0x230>
-0242 [ 5] jmp <+0x110>
-0247 [ 9] nopw 0x0(%rax,%rax,1)
-0250 [ 7] mov 0x188(%r14),%rax
+002f [ 6] mov $0x0,%r13d
+0035 [ 9] mov %fs:0x28,%rax
+003e [ 5] mov %rax,0x28(%rsp)
+0043 [ 2] xor %eax,%eax
+0045 [ 5] call <epoll_wait@plt>
+004a [ 2] test %eax,%eax
+004c [ 4] cmovns %eax,%r12d
+0050 [ 7] mov %r12d,0xc18(%r15)
+0057 [ 2] jle <+0xbb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r12d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r14 # <local-switch-0xa0,0x250,0xf0,0xa0,0xa0,0x1f0,0xb0,0xa0,0x1b0,0xb0,0xa0,0xa0,0xa0,0xa0,0xa0,0xa0,0xa0,0x1c0>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x10(%rsp)
+007f [ 1] nop
+0080 [ 4] mov 0x4(%rbx),%rsi
+0084 [ 3] mov %rsi,%rax
+0087 [ 4] shr $0x38,%rax
+008b [ 2] cmp $0x11,%al
+008d [ 2] ja <+0xa0>
+008f [ 4] movslq (%r14,%rax,4),%rax
+0093 [ 3] add %r14,%rax
+0096 [ 3] notrack jmp *%rax
+0099 [ 7] nopl 0x0(%rax)
+00a0 [ 3] mov %r15,%rdi
+00a3 [ 5] call <tomo_db0::IoLoop::fliplisten_epoll(unsigned long)>
+00a8 [ 3] add %eax,%r13d
+00ab [ 5] nopl 0x0(%rax,%rax,1)
+00b0 [ 2] inc %ebp
+00b2 [ 4] add $0xc,%rbx
+00b6 [ 3] cmp %r12d,%ebp
+00b9 [ 2] jl <+0x80>
+00bb [ 5] mov 0x28(%rsp),%rax
+00c0 [ 9] sub %fs:0x28,%rax
+00c9 [ 6] jne <+0x2e5>
+00cf [ 4] add $0x38,%rsp
+00d3 [ 3] mov %r13d,%eax
+00d6 [ 1] pop %rbx
+00d7 [ 1] pop %rbp
+00d8 [ 2] pop %r12
+00da [ 2] pop %r13
+00dc [ 2] pop %r14
+00de [ 2] pop %r15
+00e0 [ 1] ret
+00e1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00ec [ 4] nopl 0x0(%rax)
+00f0 [10] movabs $0xffffffffffff,%rax
+00fa [ 3] and %rsi,%rax
+00fd [ 2] je <+0xb0>
+00ff [ 6] mov 0x788(%rax),%edx
+0105 [ 7] mov 0x190(%r15),%rcx
+010c [ 2] cmp %edx,(%rcx)
+010e [ 2] jne <+0xb0>
+0110 [ 4] cmpb $0x0,0x33(%rax)
+0114 [ 2] jne <+0xb0>
+0116 [ 6] mov 0x784(%rax),%edx
+011c [ 3] cmp $0xfffffffe,%edx
+011f [ 2] je <+0xb0>
+0121 [ 7] mov 0x288(%r15),%rdx
+0128 [ 7] mov 0x290(%r15),%rcx
+012f [ 3] cmp %rcx,%rdx
+0132 [ 2] je <+0x152>
+0134 [11] data16 cs nopw 0x0(%rax,%rax,1)
+013f [ 1] nop
+0140 [ 3] cmp (%rdx),%rax
+0143 [ 6] je <+0xb0>
+0149 [ 4] add $0x28,%rdx
+014d [ 3] cmp %rdx,%rcx
+0150 [ 2] jne <+0x140>
+0152 [ 2] mov (%rbx),%edx
+0154 [ 6] test $0x2019,%edx
+015a [ 2] je <+0x160>
+015c [ 4] movb $0x0,0x29(%rax)
+0160 [ 3] and $0x1c,%edx
+0163 [ 2] je <+0x197>
+0165 [ 4] cmpb $0x0,0x30(%rax)
+0169 [ 5] mov %rax,0x20(%rsp)
+016e [ 2] jne <+0x197>
+0170 [ 4] movb $0x1,0x30(%rax)
+0174 [ 5] mov %rax,0x18(%rsp)
+0179 [ 5] lea 0x20(%rsp),%rax
+017e [ 7] lea 0x198(%r15),%rdi
+0185 [ 3] mov %rax,%rsi
+0188 [ 5] mov %rax,0x8(%rsp)
+018d [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
+0192 [ 5] mov 0x18(%rsp),%rax
+0197 [ 3] mov %rax,%rsi
+019a [ 3] mov %r15,%rdi
+019d [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
+01a2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01ad [ 3] nopl (%rax)
+01b0 [ 3] inc %r13d
+01b3 [ 5] jmp <+0xb0>
+01b8 [ 8] nopl 0x0(%rax,%rax,1)
+01c0 [ 7] mov 0x190(%r15),%rax
+01c7 [ 5] mov 0x10(%rsp),%rdx
+01cc [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
+01d3 [ 3] inc %r13d
+01d6 [ 2] mov (%rax),%esi
+01d8 [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
+01dd [ 5] jmp <+0xb0>
+01e2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01ed [ 3] nopl (%rax)
+01f0 [ 5] lea 0x20(%rsp),%rax
+01f5 [ 9] movq $0x0,0x20(%rsp)
+01fe [ 5] mov %rax,0x8(%rsp)
+0203 [11] data16 cs nopw 0x0(%rax,%rax,1)
+020e [ 2] xchg %ax,%ax
+0210 [ 7] mov 0xbc8(%r15),%edi
+0217 [ 5] mov 0x8(%rsp),%rsi
+021c [ 5] mov $0x8,%edx
+0221 [ 5] call <read@plt>
+0226 [ 4] cmp $0x8,%rax
+022a [ 2] je <+0x210>
+022c [ 7] mov 0x190(%r15),%rax
+0233 [ 3] inc %r13d
+0236 [ 7] incq 0x440(%rax)
+023d [ 5] jmp <+0xb0>
+0242 [11] data16 cs nopw 0x0(%rax,%rax,1)
+024d [ 3] nopl (%rax)
+0250 [ 7] mov 0x188(%r15),%rax
 0257 [ 7] movzbl 0x11460(%rax),%eax
 025e [ 2] test %al,%al
-0260 [ 6] jne <+0x110>
-0266 [ 7] mov 0xaa0(%r14),%r13d
-026d [ 3] test %r13d,%r13d
-0270 [ 6] js <+0x110>
-0276 [ 8] movl $0x0,0x18(%rsp)
-027e [ 2] xor %edx,%edx
-0280 [ 2] xor %esi,%esi
-0282 [ 5] mov $0x80800,%ecx
-0287 [ 3] mov %r13d,%edi
-028a [ 5] call <accept4@plt>
-028f [ 2] test %eax,%eax
-0291 [ 2] jns <+0x2d0>
-0293 [ 5] call <__errno_location@plt>
-0298 [ 2] mov (%rax),%eax
-029a [ 3] cmp $0x4,%eax
-029d [ 2] je <+0x27e>
-029f [ 3] cmp $0xb,%eax
-02a2 [ 2] je <+0x2b2>
-02a4 [ 7] mov 0x190(%r14),%rax
-02ab [ 7] incq 0x458(%rax)
-02b2 [ 4] mov 0x18(%rsp),%ecx
-02b6 [ 2] inc %ebp
-02b8 [ 4] add $0xc,%rbx
-02bc [ 4] add %ecx,0xc(%rsp)
-02c0 [ 3] cmp %r12d,%ebp
-02c3 [ 6] jl <+0x90>
-02c9 [ 5] jmp <+0x11f>
-02ce [ 2] xchg %ax,%ax
+0260 [ 6] jne <+0xb0>
+0266 [ 7] mov 0xaa0(%r15),%eax
+026d [ 4] mov %eax,0x8(%rsp)
+0271 [ 2] test %eax,%eax
+0273 [ 6] js <+0xb0>
+0279 [ 8] movl $0x0,0x18(%rsp)
+0281 [ 4] mov 0x8(%rsp),%edi
+0285 [ 2] xor %edx,%edx
+0287 [ 2] xor %esi,%esi
+0289 [ 5] mov $0x80800,%ecx
+028e [ 5] call <accept4@plt>
+0293 [ 2] test %eax,%eax
+0295 [ 2] jns <+0x2d0>
+0297 [ 5] call <__errno_location@plt>
+029c [ 2] mov (%rax),%eax
+029e [ 3] cmp $0x4,%eax
+02a1 [ 2] je <+0x281>
+02a3 [ 3] cmp $0xb,%eax
+02a6 [ 2] je <+0x2b6>
+02a8 [ 7] mov 0x190(%r15),%rax
+02af [ 7] incq 0x458(%rax)
+02b6 [ 4] mov 0x18(%rsp),%eax
+02ba [ 3] add %eax,%r13d
+02bd [ 5] jmp <+0xb0>
+02c2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+02cd [ 3] nopl (%rax)
 02d0 [ 5] mov $0x1,%edx
 02d5 [ 2] mov %eax,%esi
-02d7 [ 3] mov %r14,%rdi
+02d7 [ 3] mov %r15,%rdi
 02da [ 4] incl 0x18(%rsp)
 02de [ 5] call <void tomo_db0::IoLoop::admit_fd<true, false>(int, tomo_db0::UrKind)>
-02e3 [ 2] jmp <+0x27e>
+02e3 [ 2] jmp <+0x281>
 02e5 [ 5] call <__stack_chk_fail@plt>
-02ea [ 5] lea 0x20(%rsp),%r13
-02ef [ 7] lea 0x198(%r14),%rdi
-02f6 [ 5] mov %rdx,0x18(%rsp)
-02fb [ 3] mov %r13,%rsi
-02fe [ 5] call <void std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::_M_push_back_aux<tomo_db0::Client* const&>(tomo_db0::Client* const&)>
-0303 [ 5] mov 0x18(%rsp),%rdx
-0308 [ 5] jmp <+0xf2>
